### PR TITLE
Breakpad.

### PR DESCRIPTION
**Pull request by CPWstatic**
Do not support arm for now. Because the breakpad has some compiling problems at arm platform.

Release with symbols
```
package.sh -t RelWithDebInfo -r OFF -p ON -s TRUE
```

### CODE REVIEW

File: src/daemons/MetaDaemon.cpp [hunk 1]
```diff
@@ -56,9 +60,16 @@ DEFINE_bool(daemonize, true, "Whether run as a daemon process");
 
 static std::unique_ptr<apache::thrift::ThriftServer> gServer;
 static std::unique_ptr<nebula::kvstore::KVStore> gKVStore;
+#if defined(__x86_64__)
+static std::unique_ptr<google_breakpad::ExceptionHandler> gExceptionHandler;
```
Comment (Shylock-Hg) [inline src/daemons/MetaDaemon.cpp:64]: Define the static variable in `SetupBreakpad.cpp`

File: cmake/nebula/ThirdPartyConfig.cmake [hunk 2]
```diff
@@ -106,6 +106,9 @@ find_package(FLEX REQUIRED)
 find_package(LibLZMA REQUIRED)
 find_package(Fizz REQUIRED)
 find_package(Sodium REQUIRED)
+if (${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "x86_64")
+find_package(Breakpad REQUIRED)
```
Comment (yixinglu) [inline cmake/nebula/ThirdPartyConfig.cmake:110]: fix the indent

File: src/daemons/MetaDaemon.cpp [hunk 3]
```diff
@@ -198,6 +209,15 @@ int main(int argc, char* argv[]) {
     return EXIT_FAILURE;
   }
 
+#if defined(__x86_64__)
+  auto expHandler = setupBreakpad();
+  if (!expHandler.ok()) {
+    LOG(ERROR) << expHandler.status();
+    return EXIT_FAILURE;
+  }
+  gExceptionHandler = std::move(expHandler).value();
```
Comment (yixinglu) [inline src/daemons/MetaDaemon.cpp:218]: These logics could also do in the SetupBreakpad.cc

File: src/daemons/MetaDaemon.cpp (hunk 1 above)
Comment (CPWstatic) [reply src/daemons/MetaDaemon.cpp:64]: Emmm, i thought the global one should be declared in the same scope, anyway i'm ok with your idea.

File: src/daemons/StorageDaemon.cpp [hunk 4]
```diff
@@ -62,6 +66,14 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
+#if defined(__x86_64__)
+  status = setupBreakpad();
```
Comment (Shylock-Hg) [inline src/daemons/StorageDaemon.cpp:70]: Do we need this in Debug mode?

File: package/package.sh [hunk 5]
```diff
@@ -10,6 +10,8 @@
 #   -d: Whether to enable sanitizer, default OFF
 #   -t: Build type, default Release
 #   -j: Number of threads, default $(nproc)
+#   -r: Whether enable compressed debug info, defaulr ON
```
Comment (jievince) [inline package/package.sh:13]: typo error: defaulr

File: src/daemons/StorageDaemon.cpp (hunk 4 above)
Comment (CPWstatic) [reply src/daemons/StorageDaemon.cpp:70]: It's useful too, because gdb is not always available.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -12,28 +12,28 @@ defaults:
...
     strategy:
       fail-fast: false
       matrix:
         os:
-          - ubuntu1604
-          - ubuntu1804
-          - ubuntu2004
-          # - centos6
-          - centos7
+          #- ubuntu1604
```
Comment (Shylock-Hg) [inline .github/workflows/release.yml:20]: Why comment these?

Comment (CPWstatic) [reply .github/workflows/release.yml]: Just for testing.

**Review by nevermore3 — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**